code/modules/reagents/chemistry/reagents/impure_reagents.dm

```diff
@@ -16,11 +16,10 @@
 
 /datum/reagent/impurity/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
 	var/obj/item/organ/internal/liver/liver = affected_mob.get_organ_slot(ORGAN_SLOT_LIVER)
```

Comment:
```suggestion
	. = ..()
	var/obj/item/organ/internal/liver/liver = affected_mob.get_organ_slot(ORGAN_SLOT_LIVER)
```
This one looks like it lost its parent call entirely, so it would never metabolise away.